### PR TITLE
Stacklight clustering

### PR DESCRIPTION
**Pull request by thouveng**

### CODE REVIEW

**Comment by thouveng** [comment]
I still need to sperate the HAProxy definition into another file

File: classes/cluster/mk20_stacklight_advanced/stacklight/server.yml [hunk 1]
```diff
@@ -1,17 +1,44 @@
...
+    cluster_node03_name: mon03
+    cluster_node03_address: ${_param:stacklight_monitor_node03_address}
   linux:
     network:
       interface:
         ens4:
           enabled: true
           type: eth
           proto: dhcp
+  keepalived:
```
Comment (cznewt) [inline classes/cluster/mk20_stacklight_advanced/stacklight/server.yml]: remove this and include class and 2 params for keepalive

Comment (thouveng) [inline classes/cluster/mk20_stacklight_advanced/stacklight/server.yml]: Done.

File: classes/system/keepalived/cluster/instance/stacklight_monitor_vip.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+applications:
```
Comment (cznewt) [inline classes/system/keepalived/cluster/instance/stacklight_monitor_vip.yml]: this is basically service level, for this just include service keepalived and params for interface and router id [1...255] as with nova.cluster.yml for example

Comment (cznewt) [inline classes/system/keepalived/cluster/instance/stacklight_monitor_vip.yml]: Please update

**Comment by thouveng** [comment]
The problem is that we don't want to use the default 'VIP' because we may need to monitor the vrrp with a specific name. So that's why I don't use the service directly. I want something like:

> vrrp_instance stacklight_monitor_vip {
>     state BACKUP
>     nopreempt
>     garp_master_delay 5
>     garp_master_repeat 3
>     garp_master_refresh 1
>     interface ens4
>     virtual_router_id 100
>     priority 101
>     advert_int 1
>     authentication {
>         auth_type PASS
>         auth_pass password
>     }
>     virtual_ipaddress {
>         172.16.10.253
>     }
> }

File: classes/system/keepalived/cluster/instance/stacklight_monitor_vip.yml (hunk 2 above)
Comment (thouveng) [inline classes/system/keepalived/cluster/instance/stacklight_monitor_vip.yml]: See my comment in the discussion.

File: classes/system/keepalived/cluster/instance/stacklight_monitor_vip.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+- service.keepalived.support
+parameters:
+  keepalived:
+    cluster:
+      enabled: true
+      instance:
+        stacklight_monitor_vip:
+          address: ${_param:keepalived_vip_address}
+          password: ${_param:keepalived_password}
+          interface: ens4
```
Comment (cznewt) [inline classes/system/keepalived/cluster/instance/stacklight_monitor_vip.yml]: just change these to:
${_param:keepalived_stacklight_monitor_vip_address}
${_param:keepalived_stacklight_monitor_vip_interface}
${_param:keepalived_stacklight_monitor_vip_password}
interface default can go to the top of this file

Comment (thouveng) [inline classes/system/keepalived/cluster/instance/stacklight_monitor_vip.yml]: OK. Done.

File: classes/cluster/mk20_stacklight_advanced/stacklight/server.yml [hunk 4]
```diff
@@ -5,9 +5,27 @@ classes:
...
+- system.haproxy.proxy.listen.stacklight.kibana
+- system.haproxy.proxy.listen.stacklight.grafana
+- service.haproxy.proxy.single
+- system.keepalived.cluster.instance.stacklight_monitor_vip
 parameters:
   _param:
     kibana_elasticsearch_host: ${_param:stacklight_monitor_address}
+    keepalived_stacklight_monitor_vip_address: ${_param:stacklight_monitor_address}
+    keepalived_stacklight_monitor_vip_password: 'password'
+    keepalived_stacklight_monitor_vip_interface: ens4
```
Comment (cznewt) [inline classes/cluster/mk20_stacklight_advanced/stacklight/server.yml]: you can have this defaulted an system level, but ok

**Pull request merged**